streams/src/test/java/org/apache/kafka/streams/integration/AbstractResetIntegrationTest.java

```diff
@@ -265,7 +265,7 @@ public void shouldNotAllowToResetWhenIntermediateTopicAbsent() throws Exception
     public void testResetWhenLongSessionTimeoutConfiguredWithForceOption() throws Exception {
```

Comment:
Why are none of these tests...actually tests? Can you also fix this, ie add `@Test` annotations to all the tests here?

I think you can then simplify the two tests that extend this abstract test class (`ResetIntegrationTest` and `ResetIntegrationWithSslTest`) and just remove all the tests that just call `super.testXXX` -- they should automatically run all of the tests in this class

Reply:
Tried on this, partially it went good, but several tests fail under `ResetIntegrationWithSslTest`, namely: 
- `shouldNotAllowToResetWhileStreamsIsRunning`
- `shouldNotAllowToResetWhenInputTopicAbsent`
- `shouldNotAllowToResetWhenIntermediateTopicAbsent`

From output of `shouldNotAllowToResetWhileStreamsIsRunning` there are some log lines that mention SSL handshake failure:

```
    [2020-07-16 23:00:20,764] INFO stream-thread [reset-with-ssl-integration-test-not-reset-during-runtime-6971acc7-e471-4973-a652-5a09b7dce10d-StreamThread-1] State transition from PARTITIONS_ASSIGNED to RUNNING (org.apache.kafka.streams.processor.internals.StreamThread:220)
    [2020-07-16 23:00:20,766] INFO stream-client [reset-with-ssl-integration-test-not-reset-during-runtime-6971acc7-e471-4973-a652-5a09b7dce10d] State transition from REBALANCING to RUNNING (org.apache.kafka.streams.KafkaStreams:283)
    [2020-07-16 23:00:20,771] INFO [SocketServer brokerId=0] Failed authentication with /127.0.0.1 (SSL handshake failed) (org.apache.kafka.common.network.Selector:616)
    [2020-07-16 23:00:21,067] INFO [Consumer clientId=reset-with-ssl-integration-test-not-reset-during-runtime-6971acc7-e471-4973-a652-5a09b7dce10d-StreamThread-1-consumer, groupId=reset-with-ssl-integration-test-not-reset-during-runtime] Found no committed offset for partition inputTopic-0 (org.apache.kafka.clients.consumer.internals.ConsumerCoordinator:1349)
    [2020-07-16 23:00:21,093] INFO [Consumer clientId=reset-with-ssl-integration-test-not-reset-during-runtime-6971acc7-e471-4973-a652-5a09b7dce10d-StreamThread-1-consumer, groupId=reset-with-ssl-integration-test-not-reset-during-runtime] Resetting offset for partition inputTopic-0 to offset 0. (org.apache.kafka.clients.consumer.internals.SubscriptionState:397)
    [2020-07-16 23:00:21,174] INFO [SocketServer brokerId=0] Failed authentication with /127.0.0.1 (SSL handshake failed) (org.apache.kafka.common.network.Selector:616)
    [2020-07-16 23:00:21,548] INFO [SocketServer brokerId=0] Failed authentication with /127.0.0.1 (SSL handshake failed) (org.apache.kafka.common.network.Selector:616)
    [2020-07-16 23:00:21,798] INFO [SocketServer brokerId=0] Failed authentication with /127.0.0.1 (SSL handshake failed) (org.apache.kafka.common.network.Selector:616)
    [2020-07-16 23:00:22,028] INFO [SocketServer brokerId=0] Failed authentication with /127.0.0.1 (SSL handshake failed) (org.apache.kafka.common.network.Selector:616)
    [2020-07-16 23:00:22,329] INFO [SocketServer brokerId=0] Failed authentication with /127.0.0.1 (SSL handshake failed) (org.apache.kafka.common.network.Selector:616)
    [2020-07-16 23:00:22,556] INFO [SocketServer brokerId=0] Failed authentication with /127.0.0.1 (SSL handshake failed) (org.apache.kafka.common.network.Selector:616)
    [2020-07-16 23:00:22,793] INFO [SocketServer brokerId=0] Failed authentication with /127.0.0.1 (SSL handshake failed) (org.apache.kafka.common.network.Selector:616)
```

Was not able to dig deeper into what could be the reason for this. For now just left these three tests in the base class without `@Test` annotation and calling them as before in `ResetIntegrationTest` only.

Reply:
Thanks for looking into this! It definitely doesn't sound good that some of these tests don't pass with SSL. Maybe there's just some additional setup that's needed for these tests? 🤔

Anyways, we don't need to solve all that in this PR. We can revisit the issue once this is merged

Reply:
Well, actually, we don't want to run all test 2x. The Ssl test only executes 3 test (on purpose) as the other tests would be repetitive and don't add value. -- Given that we alway try to keep test execution time short, we can save about 1 minute if we avoid running the tests twice? Thoughts?

Reply:
...then why extend this abstract class at all? Or rather, why have all these tests in the abstract class and not just in the non-SSL test class?

Reply:
Not exactly sure. Guess it was just a random decision to do it this way.

Reply:
Okay, maybe I gave the wrong advise here and instead of consolidating the two classes we should have separated them. 

WDYT about just moving all but the three tests that we want to run with SSL out of `AbstractResetIntegrationTest` and into `ResetIntegrationTest` instead?

Reply:
Works for me.

Reply:
Done. Moved tests to derived class.

Also in the last commit (8218891): 
- Refactored how `testId` gets set. Instead of class-level variable it is set via `getTestId()` method which was declared as abstract in `AbstractResetIntegrationTest` (now it cannot be forgotten to be set in derived classes).
- Removed `appID` instance variable (removed instance-level state).